evm_arithmetization/src/generation/mod.rs

```diff
@@ -328,12 +425,12 @@ fn simulate_cpu<F: Field>(state: &mut GenerationState<F>) -> anyhow::Result<()>
         // Padding to a power of 2.
         state.push_cpu(row);
         row.clock += F::ONE;
-        if state.traces.clock().is_power_of_two() {
+        if (state.traces.clock() - 1).is_power_of_two() {
             break;
         }
     }
 
-    log::info!("CPU trace padded to {} cycles", state.traces.clock());
+    log::info!("CPU trace padded to {} cycles", state.traces.clock() - 1);
```

Comment:
This `-1` should be reflected in the implementation of `clock()` directly instead